webrtc-stats.html

```diff
@@ -1216,6 +1216,9 @@ <h3>
              unsigned long        framesEncoded;
              double               totalEncodeTime;
              double               averageRTCPInterval;
+             RTCQualityLimitationReason                 qualityLimitationReason;
+             record&lt;RTCQualityLimitationReason, double&gt; qualityLimitationDurations;
+             double                                     qualityTotalLimitationDuration;
```

Comment:
For the counter, I would prefer it to be a counter when the resolution or fps changes due to an adaptation reason. Ie. if we went from 15fps to 30fps or if we went from VGA to QVGA. This will be the best for detecting issues that would be visible to a user, but not visible in the stats if the quality limitations is "flapping" between two resolutions. It matters not what caused the change, cpu, bw, other for that purpose.

Reply:
So you'd prefer "unsigned long qualityLimitationChangeCount" and increment it whenever qualityLimitationReason has a new value?
Works for me - the duration can be calculated by summing across the record, if needed, while the number of transitions can't e calculated in that way.

Reply:
I don't see a way to get around the problem of "flapping" between different categories and numeric values (fps, resolution, bitrate) being continuous. What if framerate changes continuously akin to bitrate? You could try to make the categories more precise, such as "highly limited" or "slightly limited", but I don't think we can have counters for "number of times" something happens because the meaning of a high or low value of that would be implementation-specific. E.g. it might not go between 15 or 30 fps on an event, it might continuously set to every value in-between. In any case it might flap between different "reasons", or flap between limited and not limited.

Counters are problematic. Total duration works, and splitting up that duration into durations for different reasons gives you some analysis, but it does not reflect to what degree you are limited, only that you are. It explains _why_ other metrics are lower than expected, but it does not provide you the preciseness of those metrics.

If you want to know to what degree you are limited in something, or how the limitation manifests itself, like fps or resolution limitations, I think it makes more sense to poll getStats() for those metrics yourself. WDYT?

Reply:
Removing qualityTotalLimitationDuration because it is calculable from `sum of qualityLimitationDurations - qualityLimitationDuration["none"]`.